### PR TITLE
Add the missing “easy” functions for box/secretbox to libsodium-sys, and use them to implement open/seal in sodiumoxide

### PR DESCRIPTION
**Pull request by sopium**
“Completely avoid the padding dance”. Cc #138.

### CODE REVIEW

File: libsodium-sys/src/crypto_secretbox_xsalsa20poly1305.rs [hunk 1]
```diff
@@ -22,6 +22,18 @@ extern {
         clen: c_ulonglong,
         n: *const [u8; crypto_secretbox_xsalsa20poly1305_NONCEBYTES],
         k: *const [u8; crypto_secretbox_xsalsa20poly1305_KEYBYTES]) -> c_int;
+    pub fn crypto_secretbox_easy(
```
Comment (dnaq) [inline libsodium-sys/src/crypto_secretbox_xsalsa20poly1305.rs]: Shouldn't these additions be in crypto_secretbox.rs?

**Comment by sopium** [comment]
I have run cargo bench and these changes made the secretbox seal/open bench slightly faster, by about 1.8%. The box seal/open bench seems to be dominated by pubkey operations though, and no obvious speedup can be observed.

File: libsodium-sys/src/crypto_secretbox_xsalsa20poly1305.rs (hunk 1 above)
Comment (sopium) [inline libsodium-sys/src/crypto_secretbox_xsalsa20poly1305.rs]: OK. I will add a `crypto_secretbox.rs` and move these two functions there.

**Comment by sopium** [comment]
Well I run the benchmarks again and there is virtually no difference between the old and new code. I guess the vector manipulations were indeed well optimized. Kudos to @brinchj!

But at least the “easy” functions are not slower. The simplification is probably worth it.

**Pull request merged**